backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/impl/AbstractMaterialReceiptLogHandler.java

```diff
@@ -84,38 +85,35 @@ public abstract class AbstractMaterialReceiptLogHandler implements IModularContr
 	public @NonNull ExplainedOptional<LogEntryCreateRequest> createLogEntryCreateRequest(
 			@NonNull final IModularContractLogHandler.CreateLogRequest request)
 	{
-		final TableRecordReference recordRef = request.getRecordRef();
-		final I_M_InOutLine inOutLineRecord = inOutBL.getLineByIdInTrx(InOutLineId.ofRepoId(recordRef.getRecordIdAssumingTableName(getSupportedTableName())));
-		final I_M_InOut inOutRecord = inOutBL.getById(InOutId.ofRepoId(inOutLineRecord.getM_InOut_ID()));
-		final I_C_Flatrate_Term flatrateTermRecord = flatrateDAO.getById(request.getContractId());
-		final Quantity quantity = inOutBL.getQtyEntered(inOutLineRecord);
+		final InOutLineId receiptLineId = getReceiptLineId(request.getRecordRef());
+		final I_M_InOutLine receiptLineRecord = inOutBL.getLineByIdInTrx(receiptLineId);
+		final I_M_InOut receiptRecord = inOutBL.getById(InOutId.ofRepoId(receiptLineRecord.getM_InOut_ID()));
+		final I_C_Flatrate_Term contractRecord = flatrateDAO.getById(request.getContractId());
+		final Quantity quantity = inOutBL.getQtyEntered(receiptLineRecord);
 
-		final ProductId productId = ProductId.ofRepoId(inOutLineRecord.getM_Product_ID());
+		final ProductId productId = request.getProductId();
```

Comment:
This isn't ok for interim receipt logs `de.metas.contracts.modular.computing.purchasecontract.interim.MaterialReceiptLineLog`